src/Toolkit/F.php

```diff
@@ -652,11 +654,25 @@ public static function relativepath(string $file, string $in = null): string
         $file = str_replace('\\', '/', $file);
         $in   = str_replace('\\', '/', $in);
 
-        if (Str::contains($file, $in) === false) {
-            return basename($file);
+        // trim trailing slashes
+        $file = rtrim($file, '/');
+        $in   = rtrim($in, '/');
+
+        if (Str::contains($file, $in . '/') === false) {
+            // make the paths relative by stripping what they have
+            // in common and adding `../` tokens at the start
+            $fileParts = explode('/', $file);
+            $inParts = explode('/', $in);
+            while (count($fileParts) && count($inParts) && ($fileParts[0] === $inParts[0])) {
+                array_shift($fileParts);
+                array_shift($inParts);
+            }
+            
+            $parentToken = '../';
+            return str_pad('', count($inParts) * strlen($parentToken), $parentToken) . implode('/', $fileParts);
```

Comment:
Could this line be replaced with this or have I misunderstood it?

```suggestion
            return str_repeat($parentToken, count($inParts)) . implode('/', $fileParts);
```

Reply:
Absolutely, that's much clearer.